src/ipahealthcheck/ipa/host.py

```diff
@@ -20,23 +20,92 @@
 logger = logging.getLogger()
 
 
-@registry
-class IPAHostKeytab(IPAPlugin):
-    """Ensure the host keytab can get a TGT"""
-    requires = ('krb5kdc', 'dirsrv')
+class CheckKeytab(IPAPlugin):
+    service = None
+    keytab = None
 
     @duration
     def check(self):
         ccache_dir = tempfile.mkdtemp()
         ccache_name = os.path.join(ccache_dir, 'ccache')
 
         try:
-            try:
-                host_princ = str('host/%s@%s' % (api.env.host, api.env.realm))
-                kinit_keytab(host_princ, paths.KRB5_KEYTAB, ccache_name)
-            except gssapi.exceptions.GSSError as e:
-                yield Result(self, constants.ERROR,
-                             msg='Failed to obtain host TGT: %s' % e)
+            princ = str(
+                "%s/%s@%s" % (self.service, api.env.host, api.env.realm)
+            )
+            kinit_keytab(princ, self.keytab, ccache_name)
+        except gssapi.exceptions.GSSError as e:
+            yield Result(
+                self,
+                constants.ERROR,
+                msg="Failed to obtain %s TGT: %s" % (self.service, e),
+            )
         finally:
             ipautil.remove_file(ccache_name)
             os.rmdir(ccache_dir)
+
+
+@registry
+class IPAHostKeytab(CheckKeytab):
+    """Ensure the host keytab can get a TGT"""
+    requires = ('krb5kdc', 'dirsrv')
+    service = 'host'
+    keytab = paths.KRB5_KEYTAB
+
+
+@registry
+class DSKeytab(CheckKeytab):
+    """Ensure the dirsrv keytab can get a TGT"""
+    requires = ('krb5kdc', 'dirsrv')
+    service = 'ldap'
+    keytab = paths.DS_KEYTAB
+
+
+@registry
+class HTTPKeytab(CheckKeytab):
+    """Ensure the Apache keytab can get a TGT"""
+    requires = ('krb5kdc', 'dirsrv')
+    service = 'HTTP'
+    keytab = paths.HTTP_KEYTAB
+
+
+@registry
+class DNSKeytab(CheckKeytab):
+    """Ensure the DNS keytab can get a TGT"""
+    requires = ('krb5kdc', 'dirsrv')
+    service = 'DNS'
+    keytab = paths.NAMED_KEYTAB
+
+    def check(self):
+        if not os.path.exists(self.keytab):
+            return ()
+
+        return super().check()
+
+
+@registry
+class ODS_EXPORTERKeytab(CheckKeytab):
+    """Ensure the ODS exporter keytab can get a TGT"""
+    requires = ('krb5kdc', 'dirsrv')
+    service = 'DNS'
+    keytab = paths.IPA_ODS_EXPORTER_KEYTAB
+
+    def check(self):
+        if not os.path.exists(self.keytab):
+            return ()
```

Comment:
The condition could be if the host is dnssec master (with ipa config-show) rather than if the keytab exists.